packages/bson-transpilers/symbols/php/templates.yaml

```diff
@@ -0,0 +1,1214 @@
+    # functions or variables.                   #
+    #                                           #
+    # The *SymbolTemplates return names and     #
+    # usually don't take any arguments. The     #
+    # *SymbolArgsTemplates are invoked for func #
+    # calls. The first argument is always the   #
+    # lhs, i.e. the symbol returned from the    #
+    # corresponding SymbolTemplate. The rest of #
+    # the arguments are the processed arguments #
+    # passed to the original function.          #
+    #                                           #
+    #############################################
+    CodeSymbolTemplate: &CodeSymbolTemplate !!js/function >
+        () => {
+            return 'new Javascript';
+        }
+    CodeSymbolArgsTemplate: &CodeSymbolArgsTemplate !!js/function >
+        (lhs, code, scope) => {
+            return !scope 
+                ? `(${this.utils.stringifyWithSingleQuotes(code)})` 
+                : `(${this.utils.stringifyWithSingleQuotes(code)}, ${scope})`
+            ;
+        }
+    ObjectIdSymbolTemplate: &ObjectIdSymbolTemplate !!js/function >
+        () => {
+            return 'new ObjectId';
+        }
+    ObjectIdSymbolArgsTemplate: &ObjectIdSymbolArgsTemplate !!js/function >
+        (lhs, id) => {
+            return !id 
+                ? `()` 
+                : `(${this.utils.stringifyWithSingleQuotes(id)})`
+            ;
+        }
+    BinarySymbolTemplate: &BinarySymbolTemplate  !!js/function >
+        () => {
+            return 'new Binary';
+        }
+    BinarySymbolArgsTemplate: &BinarySymbolArgsTemplate !!js/function >
+        (lhs, bytes, type) => {
+            if (type === null) {
+                type = 'Binary::TYPE_GENERIC';
+            }
+            return `(${this.utils.stringifyWithSingleQuotes(bytes)}, ${type})`;
+        }
+    BinarySymbolSubtypeDefaultTemplate: &BinarySymbolSubtypeDefaultTemplate !!js/function >
+        () => {
+            return 'Binary::TYPE_GENERIC';
+        }
+    BinarySymbolSubtypeFunctionTemplate: &BinarySymbolSubtypeFunctionTemplate !!js/function >
+        () => {
+            return 'Binary::TYPE_FUNCTION';
+        }
+    BinarySymbolSubtypeByteArrayTemplate: &BinarySymbolSubtypeByteArrayTemplate !!js/function >
+        () => {
+            return 'Binary::TYPE_OLD_BINARY';
+        }
+    BinarySymbolSubtypeUuidOldTemplate: &BinarySymbolSubtypeUuidOldTemplate !!js/function >
+        () => {
+            return 'Binary::TYPE_OLD_UUID';
+        }
+    BinarySymbolSubtypeUuidTemplate: &BinarySymbolSubtypeUuidTemplate !!js/function >
+        () => {
+            return 'Binary::TYPE_UUID';
+        }
+    BinarySymbolSubtypeMd5Template: &BinarySymbolSubtypeMd5Template !!js/function >
+        () => {
+            return 'Binary::TYPE_MD5';
+        }
+    BinarySymbolSubtypeUserDefinedTemplate: &BinarySymbolSubtypeUserDefinedTemplate !!js/function >
+        () => {
+            return 'Binary::TYPE_USER_DEFINED';
```

Comment:
Question for the Compass team: what is this template actually used for? The `USER_DEFINED` constant is not intended to be used directly (although it certainly can); rather, its value signifies the start of the range where users can define their own sub-type values. Everything between 128 and 255 (inclusive) is reserved for users.

This previously came up in https://github.com/mongodb-js/compass/pull/3164#discussion_r898322747.

Reply:
@gribnoysup: question for you here.

Reply:
The current architecture of `bson-transpilers` is aiming to support everything that users can use directly, even if it's not intended for direct usage

Reply:
OK to resolve.